### ISSUE TITLE
Problem with pagination and transactions (using MSSQL)

### ISSUE THREAD

**Issue by danigb** [Vincit/objection.js#879]
Hi:

I'm running a query, and I want to add pagination. Something like this:

```js
MyModel.query(trx).where(...).orderBy(...).page(...)
```

But when running inside an actual transaction I get an error: `TransactionError: Can't acquire connection for the request. There is another request in progress.`

But it works if I remove the `page` invocation. Looking at the DEBUG output it seems like the query to get the result count it's running outside the transaction.

Any ideas?

Thanks!

**Comment by danigb** [comment]
Yes. I've read the #671 and other related issues. I agree that it's a problem with the mssql driver. But I think maybe this can be fixed inside objection. If not, do you have any workaround? Does the `get concurrency()` method works in this context?

Thanks for the incredibly fast response! 👍

**Comment by koskimas** [comment]
Actually there should be no concurrency in `page`. Which objection version are you using? You should set `concurrency` to 1 if you are using mssql although it won't help in this case.

**Comment by danigb** [comment]
Hi @koskimas,

Thanks for the response. Yesterday I've ended implementing a custom `withPagination` function that solved the issue. So instead of `query.page(...)` I use `withPagination(query, ...)` where:

```js
async function withPagination(query, page, size) {
  const offset = page * size;
  query.limit(size).offset(offset);
  return {
    results: await query,
    total: await query.clone().resultSize()
  };
}
```

...But then, I've updated objection from 1.0.0-rc2 to 1.1.3 and `page` *worked out of the box*.

Sorry to bother and thanks for the support and the awesome library.
